app/api/units_api.rb

```diff
@@ -89,13 +90,21 @@ class UnitsApi < Grape::API
                                                             :active,
                                                             :main_convenor_id,
                                                             :auto_apply_extension_before_deadline,
-                                                            :send_notifications
+                                                            :send_notifications,
+                                                            :draft_task_definition_id
                                                           )
 
       if unit.teaching_period_id.present? && unit_parameters.key?(:start_date)
         unit.teaching_period = nil
       end
 
+      if unit_parameters.key?(:draft_task_definition_id)
+        unless unit.task_definitions.exists?(unit_parameters[:draft_task_definition_id])
```

Comment:
should we also check it is a valid task? It should only have a single document upload.

Reply:
Also need to think about validating changes to tasks - if the draft learning summary is later changed it could break this.